lib/webhooks/trigger.rb

```diff
@@ -1,13 +1,67 @@
 class Webhooks::Trigger
-  def self.execute(url, payload)
-    response = RestClient::Request.execute(
+  SUPPORTED_ERROR_HANDLE_EVENTS = %w[message_created message_updated].freeze
+
+  def initialize(url, payload, webhook_type)
+    @url = url
+    @payload = payload
+    @webhook_type = webhook_type
+  end
+
+  def self.execute(url, payload, webhook_type)
+    new(url, payload, webhook_type).execute
+  end
+
+  def execute
+    perform_request
+  rescue RestClient::Exceptions::Timeout, RestClient::ExceptionWithResponse => e
+    handle_error(e)
+    Rails.logger.warn "Exception: Invalid webhook URL #{@url} : #{e.message}"
+  end
+
+  private
+
+  def perform_request
+    RestClient::Request.execute(
       method: :post,
-      url: url, payload: payload.to_json,
+      url: @url,
+      payload: @payload.to_json,
       headers: { content_type: :json, accept: :json },
       timeout: 5
     )
-    Rails.logger.info "Performed Request:  Code - #{response.code}"
-  rescue StandardError => e
-    Rails.logger.warn "Exception: invalid webhook url #{url} : #{e.message}"
+  end
+
+  def handle_error(error)
+    return unless should_handle_error?
+    return unless conversation && message
+
+    update_message_status(error)
+  end
+
+  def should_handle_error?
+    @webhook_type == :api_inbox_webhook && SUPPORTED_ERROR_HANDLE_EVENTS.include?(@payload[:event])
+  end
+
+  def update_message_status(error)
+    message.update!(status: :failed, external_error: error.message)
+  end
+
+  def conversation
```

Comment:
do we need to find the conversation here ? we have the message id right ? why not just look up for message directly

Reply:
Makes sense. Updated the PR.